modules/vm/root.tf

```diff
@@ -0,0 +1,34 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~>3.113.0"
+    }
+  }
+  required_version = ">= 1.1.0"
+}
+
+resource "azurerm_linux_virtual_machine" "server" {
+  name                = var.server_name
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  size                = "Standard_B1s"
+  admin_username      = var.admin_username
+  admin_ssh_key {
+    username   = var.admin_username
+    public_key = var.admin_ssh_public_key
+  }
+  network_interface_ids = var.network_interface_ids
+  os_disk {
+    caching              = "ReadWrite"
+    storage_account_type = "Standard_LRS"
+    disk_size_gb         = var.disk_size
+  }
+
+  source_image_reference {
```

Comment:
Am putea sa nu specificam acest block? Poate nu este nevoie de el, linux sa fie

Reply:
Da si nu. Conform documentatiei pentru [azurerm](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_virtual_machine#source_image_id) este nevoie fie de un block `source_image_reference` fie de un `source_image_id` ca sa stie dupa ce imagine sa se creeze VM-ul. As putea inlocui block-ul cu un ID doar ca personal mi s-a parut mai ok sa las block-ul in momentul in care am facut VM-ul deoarece este mai explicit.